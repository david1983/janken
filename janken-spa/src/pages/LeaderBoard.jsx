import React from 'react';
import PropTypes from 'prop-types';
import { observer, inject } from 'mobx-react';
import {
  CardContent, Card,
  Table, TableHead, TableBody, TableRow, TableCell,
} from '@material-ui/core';

const LeaderBoardPage = ({ LeaderBoard }) => (
  <div className="leaderboard-container" onLoad={() => LeaderBoard.getGames()}>
    <Card className="leaderboard-card">
      <CardContent>
        <h1>LeaderBoard</h1>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>Player name</TableCell>
              <TableCell># of Wins</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {LeaderBoard.leaderboard.map(player => (
              <TableRow>
                <TableCell>{player[0]}</TableCell>
                <TableCell>{player[1]}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </CardContent>
    </Card>

  </div>
);

LeaderBoardPage.defaultProps = {
  LeaderBoard: {},
};

LeaderBoardPage.propTypes = {
  LeaderBoard: PropTypes.object,
};

export default inject('LeaderBoard')(observer(LeaderBoardPage));
