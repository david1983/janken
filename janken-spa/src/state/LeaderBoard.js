import { observable, action } from 'mobx';
import Api from './service/api';


class LeaderBaord {
  @observable leaderboard = []

  constructor() {
    this.api = new Api('http://localhost:3001');
    this.getGames();
  }

  @action
  getGames() {
    this.api.getLeaderBoard()
      .then(result => result.data)
      .then((games) => {
        console.log(games);
        this.leaderboard = games;
      })
      .catch(console.err);
  }
}


export default LeaderBaord;
