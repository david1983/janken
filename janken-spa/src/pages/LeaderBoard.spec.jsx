import React from 'react';
import { shallow } from 'enzyme';
import LeaderBoard from './LeaderBoard';

describe('the GamePage component', () => {
  it('should match the snapshot of the shallow rendered component', () => {
    const component = <LeaderBoard />;
    const rendered = shallow(component);
    expect(rendered).toMatchSnapshot();
  });
});
