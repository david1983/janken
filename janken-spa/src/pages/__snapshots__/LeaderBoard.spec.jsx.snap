// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the GamePage component should match the snapshot of the shallow rendered component 1`] = `
<div>
  <h1>
    LeaderBoard
  </h1>
  <ul>
    <li>
      some player
    </li>
  </ul>
</div>
`;
